src/parser_urdf.cc

```diff
@@ -3588,17 +3588,18 @@ void ReduceSDFExtensionPluginFrameReplace(
 {
   std::string linkName = _link->name;
   std::string parentLinkName = _link->getParent()->name;
-  if ((*_blobIt)->FirstChildElement()->Name() == _pluginName)
+  tinyxml2::XMLElement *rootElement = (*_blobIt)->FirstChildElement();
```

Comment:
add a comment here that there is an extra root element in tinyxml2?

Reply:
ok, I think it's not an issue with the tinyxml2 API but actually a change in how we used it from #264...

Reply:
we changed what we store in the SDFExtension data structure here: https://github.com/ignitionrobotics/sdformat/pull/264/commits/6dd6d07d38871c223ff278186041df5252c024a1#diff-2bc5ca23bcfc66fe173f513399bec8065b1cb4e607d741566517d7368e6bce0fL1315-L1320

I'll update the PR description accordingly

Reply:
I just refactored the fix in https://github.com/ignitionrobotics/sdformat/pull/745/commits/b98b827670614f413b3e6a8c1633c6978ea68b19: I change the function signature so it now accepts an `XMLElement` pointer instead of an `XMLDocument` iterator. Then we can call `FirstChildElement()` once and pass the resulting pointer, which simplifies the implementation in `ReduceSDFExtensionPluginFrameReplace`

I'll fix the other functions in follow-up pull requests if you think this approach is ok

Reply:
yep the approach looks good to me.